server/src/api/v1.py

```diff
@@ -652,3 +656,45 @@ def queue_wait_time_percentiles_get():
             queue["wait_times"]
         )
     return queue_percentile_data
+
+
+def generate_token(permissions, secret_key):
+    """Generates JWT token with queue permission given a secret key"""
+    expiration_time = datetime.utcnow() + timedelta(seconds=2)
+    token_payload = {
+        "exp": expiration_time,
+        "iat": datetime.now(timezone.utc),  # Issued at time
+        "sub": "access_token",
+        "permissions": permissions,
+    }
+    token = jwt.encode(token_payload, secret_key, algorithm="HS256")
+    return token
+
+
+def validate_client_key_pair(client_id: str, client_key: str):
+    """
+    Checks client_id and key pair for validity and returns their permissions
+    """
+    client_permissions_entry = database.mongo.db.client_permissions.find_one(
```

Comment:
I hope this is not a silly question but why do the permissions need to be retrieved here and encoded in the token? I feel like the token could simply contain the `client_id`. When we actually _submit_ a job with the token in the header, the `client_id` in the token should be sufficient to retrieve the permissions from the database (just as we do here).

Also, since this is an authentication endpoint, isn't it possible that we might be using it for other purposes other than submitting a job? What I am saying is that it feels to me a bit incongruous to include permissions in response to an authentication request.